www/js/diary/list.js

```diff
@@ -170,11 +171,19 @@ angular.module('emission.main.diary.list',['ui-leaflet',
         if (angular.isDefined(userInput)) {
             // userInput is an object with data + metadata
             // the label is the "value" from the options
-            var userInputEntry = $scope.inputParams[inputType].value2entry[userInput.data.label];
-            if (!angular.isDefined(userInputEntry)) {
-              userInputEntry = ConfirmHelper.getFakeEntry(userInput.data.label);
-              $scope.inputParams[inputType].options.push(userInputEntry);
-              $scope.inputParams[inputType].value2entry[userInput.data.label] = userInputEntry;
+            var userInputEntry;
+            switch(inputType) {
+              case 'SURVEY':
+                userInputEntry = {text: userInput.data.label};
+                break;
```

Comment:
As I said earlier, we should really create directives/modules for this functionality to make it easier to re-use without editing code.